app/services/platforms/twitch.ts

```diff
@@ -35,20 +49,26 @@ export class TwitchService extends Service implements IPlatformService {
     return this.userService.platform.id;
   }
 
+  getRawHeaders(authorized = false) {
+    const map: TwitchRequestHeaders = {
+      'Client-Id': this.clientId,
+      Accept: 'application/vnd.twitchtv.v5+json',
+      'Content-Type': 'application/json',
+    };
+
+    return authorized ? { ...map, Authorization: `OAuth ${this.oauthToken}` } : map;
+  }
 
   getHeaders(authorized = false): Headers {
     const headers = new Headers();
 
-    headers.append('Client-Id', this.clientId);
-    headers.append('Accept', 'application/vnd.twitchtv.v5+json');
-    headers.append('Content-Type', 'application/json');
-
-    if (authorized) headers.append('Authorization', `OAuth ${this.oauthToken}`);
+    Object.entries(this.getRawHeaders(authorized)).forEach(([key, value]) => {
+      headers.append(key, value);
+    });
```

Comment:
is there a reason you iterate over the raw headers here instead of just dropping them in the constructor?

Reply:
`TwitchHeaders` isn't compatible with `HeadersInit`, nor does it seem to like if we cast. We'd have to do something like
```javascript
return new Headers({ ...this.getRawHeaders(authorized) })
```
which might actually not be that bad. Up to you.